### DISCUSSION TITLE
Using eloquent-sortable with `sync()`

### DISCUSSION THREAD

**Discussion by boumanb** [freekmurze/eloquent-sortable#153] [Q&A]
Hi,

Assume there's a model with a `belongsToMany` relationship. The relationship is using a custom [intermediate table class](https://laravel.com/docs/10.x/eloquent-relationships#defining-custom-intermediate-table-models) that extends `SortablePivot`. The `SortablePivot` is as follows:

```php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Relations\Pivot;
use Spatie\EloquentSortable\Sortable;
use Spatie\EloquentSortable\SortableTrait;

class SortablePivot extends Pivot implements Sortable
{
    use SortableTrait;
}
```

It often occurs that this kind of sortable relationships are being re-sorted. No new items are being added to the pivot table, just re-sorting based on user input. How would one approach this?

Currently I'm using the following method:

```php
// we have to remove all the currently linked blocks so that the order is being set correctly
$layout->blocks()->detach();

$layout->blocks()->sync($this->validated()['blocks'] ?? []);
```

I'd love to get rid of the `$layout->blocks()->detach();` while keeping the code as close as possible to what one would expect from a Laravel code base.

Ideally `$layout->blocks()->sync($this->validated()['blocks'] ?? []);` would "just work". However, the intention of the `sync()` method is syncing the ID's and not updating the pivot attributes, although inserting pivot attributes is supported. Out of the box, the `sync()` method does not mutate existing records.

**Comment by abishekrsrikaanth** [comment]
Would love to see if there are any alternative ways to not calling the `detach` method.
